diff -Naur dvbhdhomerun-20130704.orig/kernel/dvb_hdhomerun_fe.c dvbhdhomerun-20130704/kernel/dvb_hdhomerun_fe.c
--- dvbhdhomerun-20130704.orig/kernel/dvb_hdhomerun_fe.c	2016-07-13 09:01:06.485037175 +0800
+++ dvbhdhomerun-20130704/kernel/dvb_hdhomerun_fe.c	2016-07-13 08:59:47.825034000 +0800
@@ -49,7 +49,7 @@
 
 extern int hdhomerun_debug_mask;
 
-static int dvb_hdhomerun_fe_read_status(struct dvb_frontend* fe, enum fe_status* status)
+static int dvb_hdhomerun_fe_read_status(struct dvb_frontend* fe, enum fe_status *status)
 {
 	struct dvbhdhomerun_control_mesg mesg;
 	struct dvb_hdhomerun_fe_state* state = fe->demodulator_priv;
@@ -103,35 +103,27 @@
 	return 0;
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)
-static int dvb_hdhomerun_fe_get_frontend(struct dvb_frontend* fe)
-#else
-static int dvb_hdhomerun_fe_get_frontend(struct dvb_frontend* fe, struct dvb_frontend_parameters *p)
-#endif
+
+static int dvb_hdhomerun_fe_get_frontend(struct dvb_frontend *fe, struct dtv_frontend_properties *p)
 {
 	DEBUG_FUNC(1);
 	return 0;
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)
-static int dvb_hdhomerun_fe_set_frontend(struct dvb_frontend* fe)
-#else
-static int dvb_hdhomerun_fe_set_frontend(struct dvb_frontend* fe, struct dvb_frontend_parameters *p)
-#endif
+
+static int dvb_hdhomerun_fe_set_frontend(struct dvb_frontend *fe)
 {
 	struct dvb_hdhomerun_fe_state* state = fe->demodulator_priv;
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)
+
 	struct dtv_frontend_properties *p = &fe->dtv_property_cache;
-#endif
+
 
 	DEBUG_FUNC(1);
 	if (fe->ops.tuner_ops.set_params) {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)
+
 		fe->ops.tuner_ops.set_params(fe);
-#else
-		fe->ops.tuner_ops.set_params(fe,p);
-#endif
+
 		if (fe->ops.i2c_gate_ctrl)
 			fe->ops.i2c_gate_ctrl(fe, 0);
 	}
@@ -175,19 +167,11 @@
 	kfree(state);
 }
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,29)
-static int dvb_hdhomerun_fe_get_frontend_algo(struct dvb_frontend *fe)
-#else
 static enum dvbfe_algo dvb_hdhomerun_fe_get_frontend_algo(struct dvb_frontend *fe)
-#endif
 {
 	DEBUG_FUNC(1);
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,29)
-	return 1; // The two drivers that use get_grontend_algo in debian lenny return 1 for HW.
-#else
 	return DVBFE_ALGO_HW; // This is actually 0. hmm.
-#endif
 }
 
 #ifdef S2_LIPLIANIN
@@ -258,9 +242,9 @@
 }
 
 static struct dvb_frontend_ops dvb_hdhomerun_fe_ofdm_ops = {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)
+
         .delsys = { SYS_DVBT },
-#endif
+
         .info = {
                 .name                   = "HDHomeRun DVB-T",
                 .type                   = FE_OFDM,
@@ -320,9 +304,9 @@
 }
 
 static struct dvb_frontend_ops dvb_hdhomerun_fe_qam_ops = {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)
+
    .delsys = { SYS_DVBC_ANNEX_A },
-#endif
+
 	.info = {
 		.name			= "HDHomeRun DVB-C",
 		.type			= FE_QAM,
@@ -382,9 +366,9 @@
 }
 
 static struct dvb_frontend_ops dvb_hdhomerun_fe_atsc_ops = {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)
+
    .delsys = { SYS_ATSC },
-#endif
+
 	.info = {
 		.name			= "HDHomeRun ATSC",
 		.type			= FE_ATSC,
